### DISCUSSION TITLE
Implement continuous deployment by automating the release process

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#533] [Ideas]
I'm thinking we should automate our release process.  The trigger for performing an automated release via the pipeline would be a PR going into main where the aac version has been incremented beyond the latest tag.  We'd still need some way to get release notes and I'm not sure how to do that other than use the commit messages or linked issues.

The pipeline would run for each PR to main and first see if the trigger has been satisfied by running `aac version` and comparing it to the latest tag on the main branch.  If they are different, then continue with the workflow, otherwise stop.  Then just create a new release, package and apply the artifacts to the release, publish documentation, and push things to pypi.

**Comment by jondavid-black** [comment]
Here's an idea...maybe we have a rule that a PR which changes the version (and therefore triggers a release) should have the release notes as the PR description.  Then perhaps we could just copy the PR description into the release notes.

**Comment by lizzcondrey** [comment]
So, part of this has been addressed in #665 and #680. The final piece of this should be accomplished with the changelog piece in #636

**Comment by lizzcondrey** [comment]
Have merged in the changelog piece with #693, so closing this discussion thread.
